src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,372 @@
+                b.AddItem("command", cmd);
+                b.AddItem("info", sentence);
+                b.AddItem("request_id", requestid.ToString());
+                _msg.Send(b);
+            }
+            else
+            {
+                throw new InvalidOperationException("disconnected from service");
+            }
+        }
+
+        private void ResultReceived(string sender, Bundle msg)
+        {
+            Log.Debug(_tag, "OnReceived ");
+            MessageReceivedEventArgs e = new MessageReceivedEventArgs();
+            int requestid;
+            Dictionary<string, string[]> result = new Dictionary<string, string[]>();
+            result.Add("token", (string[])msg.GetItem("return_token"));
+            e.RequestId = requestid = int.Parse((string)msg.GetItem("request_id"));
+            if (msg.GetItem("command").Equals("word_tokenize"))
+            {
+                e.Message = result;
+                if (_mapsWordTokenize.ContainsKey(requestid))
+                {
+                    _mapsWordTokenize[requestid]?.Invoke(e);
+                    _mapsWordTokenize.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("pos_tag"))
+            {
+                result.Add("tag", (string[])msg.GetItem("return_tag"));
+                e.Message = result;
+                if (_mapsPosTag.ContainsKey(requestid))
+                {
+                    _mapsPosTag[requestid]?.Invoke(e);
+                    _mapsPosTag.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("ne_chunk"))
+            {
+                result.Add("tag", (string[])msg.GetItem("return_tag"));
+                e.Message = result;
+                if (_mapsNamedEntity.ContainsKey(requestid))
+                {
+                    _mapsNamedEntity[requestid]?.Invoke(e);
+                    _mapsNamedEntity.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("lemmatize"))
+            {
+                e.Message = result;
+                if (_mapsLemmatize.ContainsKey(requestid))
+                {
+                    _mapsLemmatize[requestid]?.Invoke(e);
+                    _mapsLemmatize.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("langdetect"))
+            {
+                e.Message = result;
+                if (_mapsLangDetect.ContainsKey(requestid))
+                {
+                    _mapsLangDetect[requestid]?.Invoke(e);
+                    _mapsLangDetect.Remove(requestid);
+                }
+            }
+        }
+
+        /// <summary>
+        /// An construct method  to init local env of NLP .
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public NaturalLanguageProcess()
+        {
+            _tag = Application.Current.ApplicationInfo.ApplicationId;
```

Comment:
The log tag is usually set to the module name.
App Developer can not know who is writing this log, if log tag was appid